server/public/model/integration_action_test.go

```diff
@@ -16,6 +16,209 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func TestPostAction_IsValid(t *testing.T) {
```

Comment:
We might be missing the following cases:
- Valid color for style
- Action with no id (not sure if that is a possibility at all)
- Select with invalid options (if add that functionality)

Reply:
1. Good catch, test added.
2. Yes, actions without an Id are valid. Test added.
3. See https://github.com/mattermost/mattermost/pull/30180#discussion_r1961504938